docs/src/operate/registration/auto-registration.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Entity MQTT APIs
+tags: [Child-Device, Registration]
+sidebar_position: 1
+description: Register child devices and services with MQTT
+---
+
+# Auto Registration
+
+%%te%% provides an auto-registration mechanism for clients that do not want to do an explicit registration,
+as long as they are immediate child devices of the main device or services linked to a device.
+To leverage this, they must conform to the **default topic scheme** that clearly demarcates the `device` and `service` as follows:
+
+```
+te/device/<device_id>/service/<service_id>
+```
+
+With auto-registration, when a measurement message is published to `te/device/rpi1001///m/cpu_usage` topic
+without a prior explicit registration of that entity,
+a `child-device` entity: `device/rpi1001//` is auto-registered as an immediate child device of the main device
+before the `cpu_usage` measurement is associated to it.
+
+Similarly, publishing to `te/device/rpi1002/service/collectd/m/cpu_usage` would result in the auto-registration of
+both the `child-device` entity: `device/rpi1002//` and its `service` entity: `device/rpi1002/service/collectd`.
+
+The auto-registered entities are persisted with the local MQTT broker by the `tedge-agent` with their auto-generated registration messages.
+For example, the child device `device/rpi1002//` is registered as follows:
+
+```sh te2mqtt formats=v1
+tedge mqtt pub -r 'te/device/rpi1002//' '{
+  "@type": "child-device",
+  "@parent":"device/main//",
+  "name": "rpi1002",
+}'
+```
+
+Similarly, the service: `device/rpi1002/service/collectd` is registered as follows:
+
+```sh te2mqtt formats=v1
+tedge mqtt pub -r 'te/device/rpi1002/service/collectd' '{
+  "@type": "service",
+  "@parent":"device/rpi1002//",
+  "name": "collectd",
+}'
+```
+
+:::note
+The %%te%% main device is pre-registered with the reserved default topic identifier: `device/main//`.
+:::
+
+The auto-registration feature is turned on by default and can be turned off using the following configuration setting:
+
+```sh
+tedge config set agent.entity_store.auto_register false
+```
+
+After this configuration change, the `tedge-agent` must be restarted for it to take effect.
```

Comment:
I would say then *why* this can be useful. Something along the following lines:


Turning auto-registration off let you can organize with more freedom the hierarchy of child devices and services.
- Auto-registration requires the default topic scheme to be used and works only for immediate child devices of the main device.
- With auto-registration turned off you can still use the default topic scheme (`te/device/<device_id>/service/<service_id>`) but implement a specific device hierarchy using the registration API.
- With auto-registration turned off you are even free to define a [custom schema](https://thin-edge.github.io/thin-edge.io/contribute/design/mqtt-topic-design/#using-custom-identifier-schemas).